### PR TITLE
Sets latest variant as the default

### PR DESCRIPTION
**Pull request by sdreyer**
# Description

### Select type(s) of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have fixed any merge conflicts

### CODE REVIEW

File: metadata/metadata.go [hunk 1]
```diff
@@ -1898,6 +1902,34 @@ func (serv *MetadataServer) genericCreate(ctx context.Context, res Resource, ini
...
+		parentResource = resource
+	}
+	if resource, ok := parent.(*trainingSetResource); ok {
+		resource.serialized.DefaultVariant = defaultVariant
+		parentResource = resource
+	}
+	err = serv.lookup.Set(id, parentResource)
+	if err != nil {
+		return err
+	}
```
Comment (ahmadnazeri) [inline metadata/metadata.go:1930]: missing a `return nil` at the end

Comment (sdreyer) [reply metadata/metadata.go:1930]: got it

**Review by ahmadnazeri — APPROVED** [review]

File: metadata/metadata.go [hunk 2]
```diff
@@ -1865,6 +1870,34 @@ func (serv *MetadataServer) genericCreate(ctx context.Context, res Resource, ini
 	return &pb.Empty{}, nil
 }
 
+func (serv *MetadataServer) setDefaultVariant(id ResourceID, defaultVariant string) error {
+	parent, err := serv.lookup.Lookup(id)
```
Comment (anthonylasso) [inline metadata/metadata.go:1874]: should we log this?

File: metadata/metadata.go [hunk 3]
```diff
@@ -1865,6 +1870,34 @@ func (serv *MetadataServer) genericCreate(ctx context.Context, res Resource, ini
...
+	if err != nil {
+		return err
+	}
+	var parentResource Resource
+	if resource, ok := parent.(*SourceResource); ok {
+		resource.serialized.DefaultVariant = defaultVariant
+		parentResource = resource
+	}
+	if resource, ok := parent.(*labelResource); ok {
```
Comment (anthonylasso) [inline metadata/metadata.go:1883]: should these not be else/if or switch?

**Review by anthonylasso — APPROVED** [review]

File: metadata/metadata.go (hunk 2 above)
Comment (sdreyer) [reply metadata/metadata.go:1874]: is that valuable to know?

File: metadata/metadata.go (hunk 3 above)
Comment (sdreyer) [reply metadata/metadata.go:1883]: If i do the switch I cant access the fields I want

**Pull request merged**